tree/ntuple/v7/src/RMiniFile.cxx

```diff
@@ -1099,6 +1101,14 @@ ROOT::Experimental::Internal::RNTupleFileWriter::RFileProper::ReserveBlobKey(siz
       Write(localKeyBuffer, kBlobKeyLen, offsetKey);
    }
 
+   if (keyBlob.WasAllocatedInAFreeSlot()) {
+      // If the key was allocated in a free slot, the last 4 bytes of its buffer contain the new size
+      // of the remaining free slot and we need to write it to disk before the key gets destroyed at the end of the
+      // function.
+      auto nbytesFree = *(Int_t *)(keyBlob.GetBuffer() + nbytes);
```

Comment:
Do we need to turn it to a negative value?

Reply:
No, the value in the buffer is already the correct (negative) value (see [here](https://github.com/root-project/root/blob/master/io/io/src/TKey.cxx#L514-L515))

Reply:
The value in the buffer is byte swapped.

Reply:
I see that https://github.com/root-project/root/pull/17568#discussion_r1935318321 raised the same point and was addressed.